src/hotspot/share/prims/jvmtiExport.cpp

```diff
@@ -3139,6 +3139,15 @@ bool JvmtiSampledObjectAllocEventCollector::object_alloc_is_safe_to_sample() {
     return false;
   }
 
+  // If the current thread is attaching from native and its thread oop is being
+  // allocated, things are not ready for allocation sampling.
+  if (thread->is_Java_thread()) {
```

Comment:
Nit: There is no need for this check at line 3144.
There was already check for `!thread->is_Java_thread()` and return with false at line 3138:
```
  if (!thread->is_Java_thread() || thread->is_Compiler_thread()) {
    return false;
  }
```

Reply:
+1

Reply:
Indeed, removed. Thanks.